### ISSUE TITLE
C-Gate session not ready -- unexpected message

### ISSUE THREAD

**Issue by Nivin37** [anthonywebb/homebridge-cbus#84]
Having an issue with Homebridge-cbus not connecting to C-Gate correctly, dump of issue below.

[2018-2-4 08:07:46] Loaded plugin: homebridge-cbus
[2018-2-4 08:07:46] Registering platform 'homebridge-cbus.CBus'
[2018-2-4 08:07:46] ---
[2018-2-4 08:07:46] Loaded plugin: homebridge-config-ui
[2018-2-4 08:07:46] Registering platform 'homebridge-config-ui.config'
[2018-2-4 08:07:46] ---
[2018-2-4 08:07:46] Loaded plugin: homebridge-info
[2018-2-4 08:07:46] Registering platform 'homebridge-info.Info'
[2018-2-4 08:07:46] ---
[2018-2-4 08:07:46] Loaded config.json with 0 accessories and 1 platforms.
[2018-2-4 08:07:46] ---
[2018-2-4 08:07:46] Loading 1 platforms...
[2018-2-4 08:07:46] [CBus] Initializing homebridge-cbus.CBus platform...
  cbus:client Opening connection to C-Gate… +0ms
[2018-2-4 08:07:46] Loading 0 accessories...
  cbus:client C-Gate connection open. +4ms
  cbus:client C-Gate session not ready -- unexpected message: 201 Service ready: Mosquitto C-Gate Version: v2.10.5 (build 3140) #cmd-syntax=1.0 +4ms

Looks like Homebridge is receiving an unexpected message.
anyone have any ideas? os direction I can start looking

Thanks.

Steve...

**Comment by DarylMc** [comment]
@Nivin37 
Could you mention the hardware eg where are you running CGate?
Where are you running Homebridge?

From your message I can see CGate is running.
Is your project xml in the tag folder?
Did you set project.default and project.start in the CGate config file?

**Comment by DarylMc** [comment]
I think Mosquitto in front of CGate version looks out of place.

**Comment by Nivin37** [comment]
Finally got back to this little project...
I still have a C-Bus access issue for Homebridge.
Below is a dump I get.

steve@HomeBridge:~/.homebridge$ homebridge 
[2018-11-18 10:42:14] Loaded config.json with 0 accessories and 1 platforms.
[2018-11-18 10:42:14] ---
[2018-11-18 10:42:14] Loaded plugin: homebridge-cbus
[2018-11-18 10:42:14] Registering platform 'homebridge-cbus.CBus'
[2018-11-18 10:42:14] ---
[2018-11-18 10:42:14] Loaded plugin: homebridge-config-ui
[2018-11-18 10:42:14] Registering platform 'homebridge-config-ui.config'
[2018-11-18 10:42:14] ---
[2018-11-18 10:42:14] Loaded plugin: homebridge-info
[2018-11-18 10:42:14] Registering platform 'homebridge-info.Info'
[2018-11-18 10:42:14] ---
[2018-11-18 10:42:14] Loading 1 platforms...
[2018-11-18 10:42:14] [CBus] Initializing homebridge-cbus.CBus platform...
  cbus:client Opening connection to C-Gate… +0ms
  cbus:client C-Gate connection open. +3ms
  cbus:client C-Gate session not ready -- unexpected message: 201 Service ready: Debian C-Gate Version: v2.11.1 (build 3240) #cmd-syntax=1.0 +2ms

and here is what C-Gate is reporting in the log...

20181118-104505 804 cmd33 - Host:/192.168.1.109 closed command interface from port: 37820
20181118-104507 999 sys Socket accepted.
20181118-104507 803 cmd35 - Host:/192.168.1.109 opened command interface from port: 37822
20181118-104507 899 sys Debug: New Command Context: cc044 = AccessContext Session /192.168.1.109#36
20181118-104507 766 cmd35 - Response: 201 Service ready: Debian C-Gate Version: v2.11.1 (build 3240) #cmd-syntax=1.0


Also to answer the question from above the "Debian C-Gate" is just the name I gave C-Gate in its config file when I built C-Gate.

C-Gate is working and I can control lights via openHAB.

Question: is there an issue with the C-Gate version? V2.11.1

TO me it looks to be failing at the last line,

cbus:client C-Gate session not ready...

As this is a Homebridge response anyone have any idea how I can fault find this issue???

Thanks,

Steve.
